Orange/widgets/data/owselectcolumns.py

```diff
@@ -263,53 +100,24 @@ class Outputs:
 
     def __init__(self):
         super().__init__()
+        VariablesFilter().__init__()
         self.controlArea = QWidget(self.controlArea)
         self.layout().addWidget(self.controlArea)
         layout = QGridLayout()
         self.controlArea.setLayout(layout)
         layout.setContentsMargins(4, 4, 4, 4)
         box = gui.vBox(self.controlArea, "Available Variables",
                        addToLayout=False)
-        self.filter_edit = QLineEdit()
-        self.filter_edit.setToolTip("Filter the list of available variables.")
-        box.layout().addWidget(self.filter_edit)
-        if hasattr(self.filter_edit, "setPlaceholderText"):
-            self.filter_edit.setPlaceholderText("Filter")
-
-        self.completer = QCompleter()
-        self.completer.setCompletionMode(QCompleter.InlineCompletion)
-        self.completer_model = QStringListModel()
-        self.completer.setModel(self.completer_model)
-        self.completer.setModelSorting(
-            QCompleter.CaseSensitivelySortedModel)
-
-        self.filter_edit.setCompleter(self.completer)
-        self.completer_navigator = CompleterNavigator(self)
-        self.filter_edit.installEventFilter(self.completer_navigator)
+
+        self.variables_filter = VariablesFilter()
+        box.layout().addWidget(self.variables_filter.add_filter())
 
         def dropcompleted(action):
             if action == Qt.MoveAction:
                 self.commit()
 
-        self.available_attrs = VariableListModel(enable_dnd=True)
-        self.available_attrs_proxy = VariableFilterProxyModel()
-        self.available_attrs_proxy.setSourceModel(self.available_attrs)
-        self.available_attrs_view = VariablesListItemView(
-            acceptedType=Orange.data.Variable)
-        self.available_attrs_view.setModel(self.available_attrs_proxy)
-
-        aa = self.available_attrs
-        aa.dataChanged.connect(self.update_completer_model)
-        aa.rowsInserted.connect(self.update_completer_model)
-        aa.rowsRemoved.connect(self.update_completer_model)
-
-        self.available_attrs_view.selectionModel().selectionChanged.connect(
-            partial(self.update_interface_state, self.available_attrs_view))
-        self.available_attrs_view.dragDropActionDidComplete.connect(dropcompleted)
```

Comment:
This line is removed from the widget, but not replaced in the helper function. Dragging from available attributes to the right does not trigger commit.